src/main/scala/files_metadata/application/FilesMetaUseCases.scala

```diff
@@ -228,4 +228,31 @@ class FilesMetaUseCases {
 
     repository.updateFileParent( fileUUID, newParentUUID )
   }
+
+  def unShareFile(
+      ownerUUID: UUID,
+      fileUUID: UUID,
+      otherUserUUID: UUID
+  ): Unit = {
+    val fileMeta = repository.getFileMeta( fileUUID )
+
+    if (fileMeta.ownerUuid != ownerUUID) {
+      throw DomainExceptions.FileNotOwnedException(
+        "The user does not own the file"
+      )
+    }
+    if (ownerUUID == otherUserUUID) {
+      throw DomainExceptions.FileNotOwnedException(
+        "The user and the other user are the same"
+      )
+    }
+
+    if (!repository.isFileDirectlySharedWithUser( fileUUID, otherUserUUID )) {
+      throw DomainExceptions.FileAlreadySharedException(
+        "The file is already unshared with the user"
```

Comment:
I think "The file is not shared with the given user" can be more appropriate.

Reply:
Repaired:
![image](https://github.com/hawks-atlanta/metadata-scala/assets/47455237/50a24553-07d0-424f-b113-387c325d7384)